blueprint/core/blueprint_ScrollView.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+  ==============================================================================
+
+    blueprint_ScrollView.cpp
+    Created: 10 Nov 2020 18:07pm
+
+  ==============================================================================
+*/
+
+#include "blueprint_ScrollView.h"
+
+namespace blueprint
+{
+    ScrollView::ScrollView()
+    {
+        addAndMakeVisible(viewport);
+        viewport.setScrollBarsShown(true, true);
+    }
+
+    void ScrollView::setProperty (const juce::Identifier& name, const juce::var& value)
+    {
+        View::setProperty(name, value);
+
+        if (name == overflowYProp)
```

Comment:
Hmmm thinking we can maybe shorten this up a bit too:

```
auto showX = viewport.isHorizontalScrollBarShown();
auto showY = viewport.isVerticalScrollBarShown();

if (name == overflowYProp)
{
    showY = std::invoke([](const juce::String& p) {
        if ("scroll" == p) { return true; }
        if ("hidden" == p) { return false; }
        throw std::invalid_argument("Invalid prop value. Prop \'overflow-y\' must be a string of 'hidden' or 'scroll'");
    }, props[name]);
}

if (name == overflowXProp)
{
    same...
}

// This ends up being a noop if we didn't enter the above if blocks
viewport.setScrollBarsShown(showX, showY);
```

That was the closest I could come up with for a pattern-matching like syntax, which I think would be great here. What do you think?

Reply:
Yeah nice. Maybe pull that lambda out to avoid duplication. Ok cool will take that approach.

Reply:
@nick-thompson , the only issue with that approach is if you JUST set overflow-x you'll still see the vertical scrollbar until you explicitly set overflow-y to "hidden". Do we think that's an issue?

Reply:
de-duped version looks like: 

```
        auto showX = viewport.isHorizontalScrollBarShown();
        auto showY = viewport.isVerticalScrollBarShown();

        auto overflowCheck = [](const juce::String& p, const juce::String &v)
        {
            if (juce::StringRef("scroll") == v) { return true; }
            if (juce::StringRef("hidden") == v) { return false; }

            juce::String e = "Invalid prop value. Prop '" + p + "' must be a string of 'hidden' or 'scroll'";
            throw std::invalid_argument(e.toStdString());
        };

        if (name == overflowYProp)
        {
            showY = overflowCheck(overflowYProp.toString(), value);
        }

        if (name == overflowXProp)
        {
            showX = overflowCheck(overflowXProp.toString(), value);
        }
```

Reply:
> @nick-thompson , the only issue with that approach is if you JUST set overflow-x you'll still see the vertical scrollbar until you explicitly set overflow-y to "hidden". Do we think that's an issue?

Nope, no issue by me. That makes sense to me as default behavior for the `<ScrollView>`